vulture/core.py

```diff
@@ -461,26 +501,45 @@ def _define(
         confidence=DEFAULT_CONFIDENCE,
         ignore=None,
     ):
+        def ignored(lineno):
+            if (ignore and ignore(self.filename, name)) or _match(
+                name, self.ignore_names
+            ):
+                return True
+
+            # if name not ignored, check if annotated with "# noqa"
+            if typ == "property" and sys.version_info < (3, 8):
+                # for python < 3.8, increment as many line numbers as there are
+                # decorators on the method.
+                # python3.8 onwards, lineno for property is the same as that of
+                # the decorated method.
+                lineno = lineno + len(first_node.decorator_list)
+            return lineno in (
+                self.noqa_matches[ERROR_CODES[typ]].union(
+                    self.noqa_matches["all"]
```

Comment:
I think it's faster and more readable to test the two sets separately instead of computing the union just for the test.

Reply:
That might result in the same or v/s union bug, I guess. Do you have something in mind?

Reply:
`return lineno in self.noqa_matches[ERROR_CODES[typ]] or lineno in self.noqa_matches["all"]`

Reply:
Yep, thanks!